### PR TITLE
Fix lisp status reporting issue. Make lisp use crypto/rand instead of…

### PR DESCRIPTION
**Pull request by gkodali-zededa**
… math/rand

Signed-off-by: GopiKrishna Kodali <gkodali@zededa.com>

### CODE REVIEW

File: dataplane/fib/forward.go [hunk 1]
```diff
@@ -129,6 +130,15 @@ func encryptPayload(payload []byte,
 	return true, extraLen + padLen
 }
 
+func GenerateRandToken(seed int64) int64 {
```
Comment (eriknordmark) [inline dataplane/fib/forward.go]: That's not a seed but a max.

Comment (gkodali-zededa) [reply dataplane/fib/forward.go]: Silly me. Will fix it.

**Comment by gkodali-zededa** [comment]
> What is the performance impact? I don't know how often these are called hence whether the system would run out of entropy in which case the rand calls will block.

We do not call the random number generator per packet. We call it once in the beginning and then keep incrementing the number for each packet. We re-generate a different random number when the LSB 8 bytes of IV roll over.

**Comment by eriknordmark** [comment]
And is there a separate IV generated for each peer rloc when the communication starts?

One thing which would be useful is to run a small app just calling the rand.Int in a loop and see how many calls to get a uint64 you can do before it slows down on e.g., a E100. If it  can do 10 or more that would be OK for now.

**Comment by gkodali-zededa** [comment]
> And is there a separate IV generated for each peer rloc when the communication starts?
> 
> One thing which would be useful is to run a small app just calling the rand.Int in a loop and see how many calls to get a uint64 you can do before it slows down on e.g., a E100. If it can do 10 or more that would be OK for now.

IV random generation is per ITR thread. We do not generate IV per map-cache entry. Will try what you suggest.

**Comment by eriknordmark** [comment]
@gkodali-zededa did you check on how often you can call crypto.Rand?

**Comment by gkodali-zededa** [comment]
> And is there a separate IV generated for each peer rloc when the communication starts?
> 
> One thing which would be useful is to run a small app just calling the rand.Int in a loop and see how many calls to get a uint64 you can do before it slows down on e.g., a E100. If it can do 10 or more that would be OK for now.

Erik, I have tested with the following program that loops 100 times generating int64 random number. I have timed the execution of program and saw the following time result.

real    0m1.417s
user    0m0.000s
sys     0m1.416s

Program:

```package main

import (
  "fmt"
  "crypto/rand"
  "math/big"
)

func GenerateRandToken(max int64) int64 {
  nBig, err := rand.Int(rand.Reader, big.NewInt(max))
  if err != nil {
    fmt.Printf("Random token generation for IV failed: %s", err)
  }
  n := nBig.Int64()
  return n
}

func main() {
  for i := 0; i < 100; i++ {
    myRand := uint64(GenerateRandToken(0x7fffffffffffffff))
    fmt.Printf("Rand generated is %x\n", myRand)
  }
}```

**Pull request merged**